draft-hendrickson-privacypass-public-metadata-issuance.md

```diff
@@ -43,10 +46,220 @@ TODO Abstract
+The Client then creates a TokenRequest structured as follows:
+
+~~~
+struct {
+  uint16_t token_type = 0x1234; /* Type Public Metadata Blind RSA (2048-bit) */
+  uint8_t truncated_token_key_id;
+  opaque metadata<1..2^16-1>;
+  uint8_t blinded_msg[Nk];
+} TokenRequest;
+~~~
+
+The structure fields are defined as follows:
+
+- "token_type" is a 2-octet integer, which matches the type in the challenge.
+
+- "truncated_token_key_id" is the least significant byte of the `token_key_id`
+  ({{public-issuer-configuration}}) in network byte order (in other words, the
+  last 8 bits of `token_key_id`).
+
+- "metadata" is the opaque metadata value that all `blinded_msg` values are encoded for.
+
+- "blinded_msg" is the Nk-octet request defined above.
+
+The Client then generates an HTTP POST request to send to the Issuer Request
+URI, with the TokenRequest as the content. The media type for this request
+is "application/private-token-request". An example request is shown below:
+
+~~~
+:method = POST
+:scheme = https
+:authority = issuer.example.net
+:path = /request
+accept = application/private-token-response
+cache-control = no-cache, no-store
+content-type = application/private-token-request
+content-length = <Length of TokenRequest>
+
+<Bytes containing the TokenRequest>
+~~~
+
+## Issuer-to-Client Response {#public-response}
+
+Upon receipt of the request, the Issuer validates the following conditions:
+
+- The TokenRequest contains a supported token_type.
+- The TokenRequest.truncated_token_key_id corresponds to the truncated key
+  ID of an Issuer Public Key.
+- The TokenRequest.blinded_msg is of the correct size.
+- The TokenREquest.metadata satisfies issuer requirements.
+
+If any of these conditions is not met, the Issuer MUST return an HTTP 400 error
+to the Client, which will forward the error to the client. Otherwise, if the
+Issuer is willing to produce a token token to the Client, the Issuer
+completes the issuance flow by computing a blinded response as follows:
+
+~~~
+blind_sig = BlindSign(skI, metadata, TokenRequest.blinded_msg)
+~~~
+
+The BlindSign function is defined in {{PBLINDRSA}}.
+The result is encoded and transmitted to the client in the following
+TokenResponse structure:
+
+~~~
+struct {
+  uint8_t blind_sig[Nk];
+} TokenResponse;
+~~~
+
+The Issuer generates an HTTP response with status code 200 whose content
+consists of TokenResponse, with the content type set as
+"application/private-token-response".
+
+~~~
+:status = 200
+content-type = application/private-token-response
+content-length = <Length of TokenResponse>
+
+<Bytes containing the TokenResponse>
+~~~
+
+## Finalization {#public-finalize}
+
+Upon receipt, the Client handles the response and, if successful, processes the
+content as follows:
+
+~~~
+authenticator =
+  Finalize(pkI, nonce, metadata, blind_sig, blind_inv)
+~~~
+
+The Finalize function is defined in {{PBLINDRSA}}. If this
+succeeds, the Client then constructs a Token as described in [AUTHSCHEME] as
```

Comment:
This new Token type will need a new media type, too. I wonder if we should name these something else so there's no confusion between these types and the existing ones.